### DISCUSSION TITLE
Feature request: Option to catch errors server side for backend oriented reporting

### DISCUSSION THREAD

**Discussion by tomsseisums** [vercel/next.js#49917] [App Router]
Currently it's suggested to use `error.tsx` and `useEffect` to forward errors. Not all errors should be forwarded/handled via frontend, hence I'd rather implement a custom error handler on server. It's useful when hosting off Vercel, like on GCP, to forward logs to Cloud Logging.

It seems that Instrumentation could be a good fit for this with allowing to hook to `process` events:
```
export async function register() {
  if (process.env.NEXT_RUNTIME === "nodejs") {
    process.on('uncaughtException', (err, origin) => {
        // do something
    })
  }
}
```

Another solution could be to utilize `next.config.js`, same way we get to setup app-wide redirects, we could set up app-wide error handler there:
```
const nextConfig = {
  handleServerException (err, origin) {
     // do something
  }
}

module.exports = nextConfig
```

This would give the benefit of running even on edge runtime (with caveats), because `process` is not available there.

Preferably, it should support forwarding/reformatting to frontend for graceful handling with `error.tsx`.

**Comment by devjiwonchoi** [comment]
Outstanding 👏👏 I was also wondering about this issue with error.tsx, but wasn't sure about the solution. Thank you😄

**Reply by tomsseisums** [reply]
This is a feature request, it doesn't work. :P

**Reply by devjiwonchoi** [reply]
Yeah, I meant to thank you for bringing it up😅 Anyways you might wanna create an issue for this, which you can do it [here](https://github.com/vercel/next.js/issues/new/choose).
